### PR TITLE
Add scores (precision, recall, center and radius mad)

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
(still need to add some tests, and the actual ScoreType classes)

### CODE REVIEW

File: workflow/scores/precision_recall.py [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+
+    idxs_true, idxs_pred = linear_sum_assignment(1 - iou_matrix)
+
+    if (not idxs_true.size) or (not idxs_pred.size):
+        ious = np.array([])
+    else:
+        ious = iou_matrix[idxs_true, idxs_pred]
+    return idxs_true, idxs_pred, ious
+
+
```
Comment (kegl) [inline workflow/scores/precision_recall.py:47]: I'm wondering if we couldn't use an extension of this for combining predictions. Remember, the specs are:

1. it gets a list of y_preds (list of list of disks),
2. it finds matching groups (each group contains at most one disk of each y_pred)
3. it makes them "vote": only those disks get outputted which contain a member disk from at least half of the y_preds
4. it averages x, y, and r in each group.

Comment (jorisvandenbossche) [reply workflow/scores/precision_recall.py:47]: yes, I think we can reuse this code for the prediction class (the above indeed seems a sensible way to combine predictions)

**Comment by jorisvandenbossche** [comment]
@aboucaud can you give this a look?

File: workflow/scores/precision_recall.py [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+        ious : corresponding IOU value of each match
+
+        The length of the 3 arrays is identical and the minimum of the length
+        of y_true and y_pred
+
+    """
+    n = len(y_true)
+    m = len(y_pred)
+
+    iou_matrix = np.empty((n, m))
```
Comment (aboucaud) [inline workflow/scores/precision_recall.py]: Maybe `n_true` / `n_pred` would be clearer than `n` / `m`

File: workflow/tests/test_scores.py [hunk 3]
```diff
@@ -45,3 +50,38 @@ def test_ospa():
...
+    assert mad_radius(y_true, y_pred) == pytest.approx(0.1)
+    assert mad_center(y_true, y_pred) == pytest.approx(0.1)
+
+    # no match
+    y_true = [[(1, 1, 1)]]
+    y_pred = [[(3, 3, 1)]]
+    assert precision(y_true, y_pred) == 0
+    assert recall(y_true, y_pred) == 0
+    assert math.isnan(mad_radius(y_true, y_pred))
+    assert math.isnan(mad_center(y_true, y_pred))
```
Comment (aboucaud) [inline workflow/tests/test_scores.py]: Add newline at the end of file 😄

**Review by aboucaud — COMMENTED** [review]
LGTM

**Pull request merged**